packages/eservice-template-updater/src/eserviceTemplateUpdaterConsumerServiceV2.ts

```diff
@@ -0,0 +1,419 @@
+      { type: "EServiceTemplateDeleted" },
+      { type: "EServiceTemplateDraftUpdated" },
+      { type: "EServiceTemplateDraftVersionDeleted" },
+      { type: "EServiceTemplateDraftVersionUpdated" },
+      { type: "EServiceTemplateRiskAnalysisAdded" },
+      { type: "EServiceTemplateRiskAnalysisDeleted" },
+      { type: "EServiceTemplateRiskAnalysisUpdated" },
+      { type: "EServiceTemplateVersionActivated" },
+      { type: "EServiceTemplateVersionAdded" },
+      { type: "EServiceTemplateVersionInterfaceAdded" },
+      { type: "EServiceTemplateVersionInterfaceDeleted" },
+      { type: "EServiceTemplateVersionInterfaceUpdated" },
+      { type: "EServiceTemplateVersionPublished" },
+      { type: "EServiceTemplateVersionSuspended" },
+      () => Promise.resolve
+    )
+    .exhaustive();
+}
+
+function getTemplateFromEvent(
+  msg: EServiceTemplateEventEnvelope
+): EServiceTemplateV2 {
+  if (!msg.data.eserviceTemplate) {
+    throw missingKafkaMessageDataError("eserviceTemplate", msg.type);
+  }
+
+  return msg.data.eserviceTemplate;
+}
+
+function getTemplateVersionFromEvent(
+  msg: EServiceTemplateEventEnvelope & {
+    data: { eserviceTemplateVersionId: string };
+  }
+): EServiceTemplateVersionV2 {
+  const eserviceTemplate = getTemplateFromEvent(msg);
+
+  const eserviceTemplateVersion = eserviceTemplate.versions.find(
+    (v) => v.id === msg.data.eserviceTemplateVersionId
+  );
+
+  if (!eserviceTemplateVersion) {
+    throw missingKafkaMessageDataError("eserviceTemplateVersion", msg.type);
+  }
+
+  return eserviceTemplateVersion;
+}
+
+async function commitUpdateToInstanceDescriptors(
+  msg: EServiceTemplateEventEnvelope & {
+    data: { eserviceTemplateVersionId: string };
+  },
+  refreshableToken: RefreshableInteropToken,
+  correlationId: CorrelationId,
+  action: (
+    eservice: catalogApi.EService,
+    descriptor: catalogApi.EServiceDescriptor,
+    headers: InteropHeaders
+  ) => Promise<void>
+): Promise<void> {
+  await commitUpdateToInstances(
+    msg,
+    refreshableToken,
+    correlationId,
+    async (instance, headers) => {
+      const instanceDescriptor = instance.descriptors.find(
+        (d) => d.templateVersionId === msg.data.eserviceTemplateVersionId
+      );
+
+      if (instanceDescriptor) {
+        await action(instance, instanceDescriptor, headers);
+      }
+    }
+  );
+}
+
+async function commitUpdateToInstances(
+  msg: EServiceTemplateEventEnvelope,
+  refreshableToken: RefreshableInteropToken,
+  correlationId: CorrelationId,
+  action: (
+    eservice: catalogApi.EService,
+    headers: InteropHeaders
+  ) => Promise<void>
+): Promise<void> {
+  const token = (await refreshableToken.get()).serialized;
+  const headers = getInteropHeaders({
+    token,
+    correlationId,
+  });
+
+  const instances = await getAllEServiceTemplateInstances(
+    getTemplateFromEvent(msg).id,
+    headers
+  );
+
+  await Promise.all(instances.map((instance) => action(instance, headers)));
```

Comment:
code smell: this is technically fine, but for templates that are used to create thousands of eservices (like the ones for Comuni), sending so many requests together may impact the destination microservice.
Consider chunking the requests